### PR TITLE
#13: added /quote functionality

### LINKED ISSUE #13: add /quote command to quote specific messages
I guess it should work like the $quote command of another bot - you enter /quote and an message link as a required argument, and the bot will post the message as an embed. 
Main benefit would be that quoted messages are much easier to read than just pasting a message link into the chat.

### PR DESCRIPTION
**Pull request by Jonas164**
also refactored image attachements in embeds

Sorry for formatting changes, I don't know where they are coming from :(

Closes #13

### CODE REVIEW

File: src/main/kotlin/apoy2k/greenbookbot/listener/CommandListener.kt [hunk 1]
```diff
@@ -97,6 +106,48 @@ class CommandListener(
...
+
+            event.replyEmbeds(builder.build()).await()
+        }
+    }
+
+
+    private suspend fun retrieveMessage(event: SlashCommandInteractionEvent, channelId: String, messageId: String): Message? {
+        val channel = event.jda.getTextChannelById(channelId)
+        val message = channel?.retrieveMessageById(messageId)?.await()
+        log.info("Retrieved message: ${message?.contentDisplay}")
```
Comment (ApoY2k) [inline src/main/kotlin/apoy2k/greenbookbot/listener/CommandListener.kt:141]: Seems a bit excessive to log the whole message?

File: src/main/kotlin/apoy2k/greenbookbot/listener/CommandListener.kt [hunk 2]
```diff
@@ -97,6 +106,48 @@ class CommandListener(
...
+        return message
+    }
+
+    private suspend fun getMessageAuthor(message: Message): Member? {
```
Comment (ApoY2k) [inline src/main/kotlin/apoy2k/greenbookbot/listener/CommandListener.kt:145]: I feel this doesn't really add any value as a method 🤔

File: src/main/kotlin/apoy2k/greenbookbot/listener/CommandListener.kt [hunk 3]
```diff
@@ -97,6 +106,48 @@ class CommandListener(
...
+
+            attachImageToBuilder(builder, message)
+
+            event.replyEmbeds(builder.build()).await()
+        }
+    }
+
+
+    private suspend fun retrieveMessage(event: SlashCommandInteractionEvent, channelId: String, messageId: String): Message? {
+        val channel = event.jda.getTextChannelById(channelId)
```
Comment (ApoY2k) [inline src/main/kotlin/apoy2k/greenbookbot/listener/CommandListener.kt:139]: Not sure, but check if this works with threads as well? (see postFav method)

**Review by ApoY2k — CHANGES_REQUESTED** [review]

File: src/main/kotlin/apoy2k/greenbookbot/listener/CommandListener.kt (hunk 1 above)
Comment (Jonas164) [reply src/main/kotlin/apoy2k/greenbookbot/listener/CommandListener.kt]: Fixed

File: src/main/kotlin/apoy2k/greenbookbot/listener/CommandListener.kt (hunk 2 above)
Comment (Jonas164) [reply src/main/kotlin/apoy2k/greenbookbot/listener/CommandListener.kt]: Inlined and Removed

File: src/main/kotlin/apoy2k/greenbookbot/listener/CommandListener.kt (hunk 3 above)
Comment (Jonas164) [reply src/main/kotlin/apoy2k/greenbookbot/listener/CommandListener.kt]: Fixed!

**Review by ApoY2k — APPROVED** [review]

**Pull request merged**